packages/cw721/src/testing/unit_tests.rs

```diff
@@ -0,0 +1,1185 @@
+            royalty_info: Some(RoyaltyInfo {
+                payment_address: Addr::unchecked("payment_address"),
+                share: Decimal::percent(MAX_ROYALTY_SHARE_PCT)
+                    .to_string()
+                    .parse()
+                    .unwrap(),
+            }),
+        };
+        let updated_collection_info_msg = CollectionInfoMsg {
+            name: Some("new_collection_name".into()),
+            symbol: Some("new_collection_symbol".into()),
+            extension: updated_extension_msg,
+        };
+        let info_other = mock_info(OTHER_ADDR, &[]);
+        let err = contract
+            .execute(
+                deps.as_mut(),
+                env.clone(),
+                info_other,
+                Cw721ExecuteMsg::UpdateCollectionInfo {
+                    collection_info: updated_collection_info_msg,
+                },
+            )
+            .unwrap_err();
+        assert_eq!(
+            err,
+            Cw721ContractError::Ownership(cw_ownable::OwnershipError::NotOwner)
+        );
+    }
+}
+
+#[test]
+fn use_metadata_extension() {
+    let mut deps = mock_dependencies();
+    let contract = Cw721Contract::<
+        DefaultOptionMetadataExtension,
+        MetadataMsg,
+        DefaultOptionCollectionInfoExtension,
+        CollectionInfoExtensionMsg<RoyaltyInfo>,
+        Empty,
+    >::default();
+
+    let info = mock_info(CREATOR_ADDR, &[]);
+    let init_msg = Cw721InstantiateMsg {
+        name: "collection_name".into(),
+        symbol: "collection_symbol".into(),
+        collection_info_extension: None,
+        minter: None,
+        creator: None,
+        withdraw_address: None,
+    };
+    let env = mock_env();
+    contract
+        .instantiate(
+            deps.as_mut(),
+            env.clone(),
+            info.clone(),
+            init_msg,
+            "contract_name",
+            "contract_version",
+        )
+        .unwrap();
+
+    let token_id = "Enterprise";
+    let token_uri = Some("https://starships.example.com/Starship/Enterprise.json".into());
+    let extension = Some(Metadata {
+        description: Some("Spaceship with Warp Drive".into()),
+        name: Some("Starship USS Enterprise".to_string()),
+        ..Metadata::default()
+    });
+    let exec_msg = Cw721ExecuteMsg::Mint {
+        token_id: token_id.to_string(),
+        owner: "john".to_string(),
+        token_uri: token_uri.clone(),
+        extension: extension.clone(),
+    };
+    contract
+        .execute(deps.as_mut(), env.clone(), info, exec_msg)
+        .unwrap();
+
+    let res = contract
+        .query_nft_info(deps.as_ref(), env, token_id.into())
+        .unwrap();
+    assert_eq!(res.token_uri, token_uri);
+    assert_eq!(res.extension, extension);
+}
+
+#[test]
+fn test_migrate() {
+    let mut deps = mock_dependencies();
+
+    let env = mock_env();
+    use cw721_base_016 as v16;
+    v16::entry::instantiate(
```

Comment:
migration test and especially how minter, creator, collection info looks like BEFORE and AFTER migration